archivessnake/edit_notes.py

```diff
@@ -78,20 +76,18 @@ def get_parser():
     parser.add_argument("-r", "--replace_string", help="The new note content to replace the old note content. (Only relevant if you are modifying note(s))")
     return parser
 
-def main():
+def main(client, writer):
     """Main function, which is run when this script is executed"""
     start_time = time.time()
     parser = get_parser()
     args = parser.parse_args()
-    global aspace
-    aspace = ASpace()
-    global writer
-    writer = csv.writer(open(spreadsheet_path, "w"))
-    create_spreadsheet(["Box Number", "Archival Object URI"])
-    process_tree(args, aspace.repositories(AS_REPOSITORY).resources(args.resource_id))
+    create_spreadsheet(["Collection Title", "Finding Aid Number", "URI", "Ref ID", "Object Title", "Box Number"])
+    process_tree(args, client.get(f'/repositories/2/resources/{(args.resource_id)}').json())
     elapsed_time = time.time() - start_time
     print("Time Elapsed: " + time.strftime("%H:%M:%S", time.gmtime(elapsed_time)))
 
-
 if __name__ == "__main__":
-    main()
+    client = ASpace().client
+    spreadsheet_path = os.path.join(os.path.abspath(os.path.dirname(__file__)), OUTPUT_FILENAME)
+    writer = csv.writer(open(spreadsheet_path, "w"))
+    main(client, writer)
```

Comment:
I would suggest moving all of this up to the `main` method - there's no reason to do some instantiation here and it just forces you to have to pass those arguments up to main.